leases/schema/mutations.py

```diff
@@ -356,6 +357,34 @@ def mutate_and_get_payload(cls, root, info, **input):
         return DeleteWinterStorageLeaseMutation()
 
 
+class AssignNewStickerNumberMutation(graphene.ClientIDMutation):
+    class Input:
+        lease_id = graphene.String(required=True)
+
+    new_sticker_number = graphene.Int()
+
+    @classmethod
+    @change_permission_required(WinterStorageLease)
+    @transaction.atomic
+    def mutate_and_get_payload(cls, root, info, **input):
+        lease = get_node_from_global_id(
+            info,
+            input.pop("lease_id"),
+            only_type=WinterStorageLeaseNode,
+            nullable=False,
+        )
+        if lease.status != LeaseStatus.PAID:
+            raise VenepaikkaGraphQLError(_("Lease must be in PAID status"))
+        elif lease.application.area_type != ApplicationAreaType.UNMARKED:
+            raise VenepaikkaGraphQLError(_("Lease must refer to unmarked area"))
```

Comment:
We should add this check on the model's `clean()` instead, to avoid that stickers could be assigned to "marked" applications in any other way.

Reply:
Agreed to keep it on the service level rather than the model level.